File: src/main/java/j/examples/validators/condition/NameValidator.java
package j.examples.validators.condition;

public class NameValidator {

    private static final int MINIMAL_LENGTH = 1;
    private static final int MAXIMAL_LENGTH = 64;

    boolean validate(String name) {
        if (name.length() < MINIMAL_LENGTH || name.length() > MAXIMAL_LENGTH) {
            return false;
        }

        if (Character.isWhitespace(name.charAt(0)) || Character.isWhitespace(name.charAt(name.length() - 1))) {
            return false;
        }

        return true;
    }
}

File: src/main/kotlin/k/examples/validators/condition/NameValidator.kt
package k.examples.validators.condition


class NameValidator {
    fun validate(name: String): Boolean {
        if (name.length < MINIMAL_LENGTH || name.length > MAXIMAL_LENGTH) {
            return false
        }

        if (Character.isWhitespace(name[0]) || Character.isWhitespace(name[name.length - 1])) {
            return false
        }

        return true;
    }

    companion object {
        private const val MINIMAL_LENGTH = 1
        private const val MAXIMAL_LENGTH = 64
    }
}
